File: vinylandbeats-app/src/app/shared/models/product.ts
export interface Product {
  id: number;
  name: string;
  description: string;
  price: number;
  stock: number;
  imageUrl?: string | null;
  categoryId: number;
  categoryName: string;
  sellerId: string | null;
  sellerName: string;
  tags: Tag[];
}

export interface Category {
  id: number;
  name: string;
}

export interface Tag {
  id: number;
  name: string;
}

export interface CurrentUser {
  id: string;
  name: string;
  email: string;
  roles: string[];
}

export interface CreateProductDto {
  name: string;
  description: string;
  price: number;
  stock: number;
  imageUrl?: string | null;
  categoryId: number;
  tagIds?: number[];
}

export interface UpdateProductDto {
  name: string;
  description: string;
  price: number;
  stock: number;
  imageUrl?: string | null;
  categoryId: number;
  tagIds?: number[];
}

File: VinylAndBeats/Models/Product.cs
using System.ComponentModel.DataAnnotations;

namespace VinylAndBeats.Models;

public class Product : BaseEntity
{
    [Required]
    [MinLength(2)]
    public string Name { get; set; } = string.Empty;

    [Required]
    public string Description { get; set; } = string.Empty;

    [Range(0.01, 1000000)]
    public decimal Price { get; set; }

    [Range(1, 100000)]
    public int Stock { get; set; }

    public string? ImageUrl { get; set; }

    public int CategoryId { get; set; }
    public Category? Category { get; set; }

    public string? SellerId { get; set; }
    public ApplicationUser? Seller { get; set; }

    public List<Tag> Tags { get; set; } = new();
}